### PR TITLE
Added closed_loop option in DomainIntegralAction

### LINKED ISSUE #10299: Add closed_loop option in DomainIntegralAction
## Rationale

The ``closed_loop`` option is provided in ``CrackFrontDefinition``.  However, the ``DomainIntegralAction`` does not set this parameter when creating the ``CrackFrontDefinition`` and thus the default value of ``closed_loop = false`` is always used by ``CrackFrontDefinition``.  

## Description

The ``closed_loop`` option tells the ``CrackFrontDefinition`` the set of crack front points form a loop and the first and last point will be connected. With this fix, the segment which connects the first and last point will be correct.

## Impact

Might affect a few tests which use DomainIntegralAction

### PR DESCRIPTION
**Pull request by jiangwen84**
1. added closed_loop option in DomainIntegralAction
2. addressed one minor comment which is pointed out in PR#10267
3. revised penny_crack_cfp.i for regression test
4. fixed one typo in xfem/test/tests/crack_tip_enrichement -> crack_tip_enrichment
close #10299

### CODE REVIEW

File: modules/tensor_mechanics/src/actions/DomainIntegralAction.C [hunk 1]
```diff
@@ -304,8 +302,11 @@ DomainIntegralAction::act()
     if (_use_crack_front_points_provider)
     {
       params.set<UserObjectName>("crack_front_points_provider") = _crack_front_points_provider;
-      params.set<unsigned int>("number_points_from_provider") = _number_crack_front_points;
+      params.set<unsigned int>("number_points_from_provider") =
+          getParam<unsigned int>("number_points_from_provider");
     }
+    if (_closed_loop)
```
Comment (permcody) [inline modules/tensor_mechanics/src/actions/DomainIntegralAction.C]: You just introduced another error: Up above you conditionally initialized `_closed_loop`, that means that this statement works half the time and returns garbage the other half of the time. 

I don't immediately see where the parameter closed_loop is coming from, perhaps it's already in the code base and is being inherited? It wasn't added in this PR. Can you just provide a default value for that parameter and **always** initialize the class member? Less code and bug free ;)

File: modules/tensor_mechanics/src/actions/DomainIntegralAction.C [hunk 2]
```diff
@@ -304,8 +302,11 @@ DomainIntegralAction::act()
     if (_use_crack_front_points_provider)
     {
       params.set<UserObjectName>("crack_front_points_provider") = _crack_front_points_provider;
-      params.set<unsigned int>("number_points_from_provider") = _number_crack_front_points;
+      params.set<unsigned int>("number_points_from_provider") =
+          getParam<unsigned int>("number_points_from_provider");
```
Comment (permcody) [inline modules/tensor_mechanics/src/actions/DomainIntegralAction.C]: You might use the following method to clean up this block of code:
https://github.com/idaholab/moose/blob/devel/framework/include/utils/InputParameters.h#L576-L577

**Review by permcody — CHANGES_REQUESTED** [review]

File: modules/tensor_mechanics/src/actions/DomainIntegralAction.C (hunk 1 above)
Comment (jiangwen84) [reply modules/tensor_mechanics/src/actions/DomainIntegralAction.C]: Good point. Thanks!

File: modules/tensor_mechanics/src/actions/DomainIntegralAction.C (hunk 2 above)
Comment (jiangwen84) [reply modules/tensor_mechanics/src/actions/DomainIntegralAction.C]: Nice tool. Thanks!

File: modules/tensor_mechanics/src/actions/DomainIntegralAction.C [hunk 3]
```diff
@@ -135,9 +135,9 @@ DomainIntegralAction::DomainIntegralAction(const InputParameters & params)
     mooseError("DomainIntegral error: invalid q_function_type.");
 
   if (isParamValid("crack_front_points"))
-  {
     _crack_front_points = getParam<std::vector<Point>>("crack_front_points");
-  }
+  if (isParamValid("closed_loop"))
+    _closed_loop = getParam<bool>("closed_loop");
```
Comment (permcody) [inline modules/tensor_mechanics/src/actions/DomainIntegralAction.C]: This line can just be moved to the initialization list now. Just trying to be tidy. I don't see the point in initializing to false only to change the value again. Just set it to the desired value directly.

**Review by permcody — APPROVED** [review]

**Pull request merged**